### ISSUE TITLE
Parameters and return part of detaileddescription in XML

### ISSUE THREAD

**Issue by rpav** [doxygen/doxygen#6919]
Having `parameterlist` in `detaileddescription` makes custom formatting and processing harder, especially when the position in the block is based on where it is in the *comment*.  This is mostly redundant, anyway... at least if `parameterdescription` were part of the `param` block.

**Comment by albert-github** [comment]
A bit clearer description would help including some places where the problem occurs.

So can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.
Please also specify the doxygen version used.

**Comment by rpav** [comment]
[doxy-example.tar.gz](https://github.com/doxygen/doxygen/files/3060061/doxy-example.tar.gz)

The config is default generated, comments removed, all generates but XML turned off.

Specifically, from the generated `test_8hpp.xml`:
```xml
<briefdescription>
<para>This is brief. </para>
        </briefdescription>
        <detaileddescription>
<para><parameterlist kind="param"><parameteritem>
<parameternamelist>
<parametername>p0</parametername>
</parameternamelist>
<parameterdescription>
<para>First </para>
</parameterdescription>
</parameteritem>
```
The desire is to format `parameterlist` separately and not have this as part of the description.  Really, there doesn't seem to be any new information here besides `parameterdescription` which seems more ideally placed in the existing `param` block:

```xml
        <param>
          <type>int</type>
          <declname>p0</declname>
        </param>
```

**Comment by albert-github** [comment]
So one would get e.g.:
```
<briefdescription>...</briefdescription>
<parameterlist>...</parameterlist>
<detaileddescription>...</detaileddescription>
```
as is doesn't sound unreasonable, but
- I don't know the reason why the current implementation has been chosen, but probably because in HTML (and other outputs) the parameterlist is also part of the detailed description
- will be hard to change as it would break a lot of existing software that uses the xml output (consistency!).

**Comment by rpav** [comment]
Actually two things ought to keep this compatible,

1. Add relevant description/doc string to `<param>`
2. Have a config option to disable `<parameterlist>` in `<detaildescription>`, which can remain OFF by default

**Comment by albert-github** [comment]
So not creating a separate `<parameterlist>` but placing all information with `<param>`, at first sight would make sense, I don't know by head what other kind is information is placed in the `<parameterlist>` part, but would probably not be problematic to have everything with `<param>`.

Extra configuration option: ouch one more, there are already that many and people don't know what is what to obtain the right output / the output they want.

**Comment by rpav** [comment]
Yeah I figured a config option would be like "another one?!" or "well, there are already a zillion, why not another" .. it would actually be nice if there was more XML customization in general, but I don't know the best way here.

**Comment by albert-github** [comment]
More XML customization could probably not hurt, but for this a concept should be drawn  so not every ad hoc change has to be fitted in (note this is no criticism to your request but more to get a general picture). When everything would be, reasonable, clear is would also be easier to add one or more configuration options covering all the possibilities.
